File: lib/leetcode/1137-n-th-tribonacci-number/solution_2.go

// 0
// ms
// Beats
// 100.00%

func tribonacci(n int) int {
	switch n {
	case 0:
		return 0
	case 1:
		return 1
	case 2:
		return 1
	}

	t0 := 0
	t1 := 1
	t2 := 1

	for i := 3; i < n+1; i++ {
		t2, t1, t0 = t2+t1+t0, t2, t1
	}

	return t2
}

File: lib/leetcode/155-min-stack/solution_2.dart
// 20
// ms
// Beats
// 87.88%

class MinStack {
  MinStack() {}

  final stack = <int>[];
  final min_stack = <int>[];

  void push(int value) {
    stack.add(value);

    if (min_stack.isEmpty) {
      min_stack.add(value);
    } else {
      if (value <= min_stack.last) min_stack.add(value);
    }
  }

  void pop() {
    if (min_stack.last == stack.removeLast()) min_stack.removeLast();
  }

  int top() {
    return stack.last;
  }

  int getMin() {
    return min_stack.last;
  }
}

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack obj = MinStack();
 * obj.push(value);
 * obj.pop();
 * int param3 = obj.top();
 * int param4 = obj.getMin();
 */
